gtsfm/loader/yfcc_imb_loader.py

```diff
@@ -129,7 +125,10 @@ def is_valid_pair(self, idx1: int, idx2: int) -> bool:
         Returns:
             validation result.
         """
-        return (idx1, idx2) in self._image_pairs
+        if not super().is_valid_pair(idx1, idx2):
+            return False
+
+        return (idx1, idx2) in self._image_pairs or (idx2, idx1) in self._image_pairs
```

Comment:
why do we need both ways now?

Reply:
because IMB dataset does not have any implied ordering.

A pair between image (1, 5) can be either stored as (1, 5) or (5, 1) in IMB, but not both. I'll modify the logic to just flip the pairs in the initialization step, so that this confusing check if not needed.